src/main/java/org/elasticsearch/cluster/InternalClusterInfoService.java

```diff
@@ -140,6 +151,11 @@ public void onMaster() {
         }
     }
 
+    // called from tests as well
+    void updateOnce() {
```

Comment:
This needs actual javadocs, just looking at it is doesn't tell that it executes the job with rescheduling false (you have to read the constructor argument for `ClusterInfoUpdateJob` to see that